eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/UseCaughtExceptionCauseCheck.java

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Ensures caught exceptions are included as exception cause in subsequently thrown exception.
```

Comment:
Can you please add a bit more commentary here as well as sample code to demonstrate exactly what your expectations are for this check? Thanks!

Reply:
updated.